File: LC/lc238/lc238.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function(nums) {
  let cur = 1, tmp;
  let res = Array(nums.length).fill(1);
  for (i = nums.length - 1; i >= 0; i--){
      res[i] *= cur;
      cur *= nums[i];
  }
  cur = 1;
  for (i = 0; i < nums.length; i++){
      res[i] *= cur;
      cur *= nums[i];
  }
  return res;
};

File: LC/lc238/lc238.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        L = len(nums)
        res = [1]*L
        prod = 1
        for i in range(1, L):
            prod *= nums[i-1]
            res[i] *= prod
        prod = 1
        for i in range(L-1, 0, -1):
            prod *= nums[i]
            res[i-1] *= prod
        return res
        